### PR TITLE
Containerize Enclave manager and Shell as services

### PR DESCRIPTION
**Pull request by manju956**
- A separate container service is added for enclave manager and its own Dockerfile is newly created.
- Previously available tcf service is renamed to avalon-shell to execute test cases and run clients.
- All references of TrustedComputeFrameWork has been renamed to avalon.
- A separate container service is created for shell to execute tests and run clients.

Signed-off-by: manju956 <manjunath.a.c@intel.com>

### CODE REVIEW

File: config/tcs_config.toml [hunk 1]
```diff
@@ -40,7 +40,8 @@ max_work_order_count = 10
 # Service Provider ID (SPID) is a 32-digit hex string tied to the
 # enclave implementation. Replace dummy SPID with value obtained after
 # subscription to run TCS in Intel SGX HW mode.
-spid = "DEADBEEF00000000DEADBEEF00000000"
+#spid = "DEADBEEF00000000DEADBEEF00000000"
+spid = "F96B1426FEC20F217482D3C9C31FC7EA"
```
Comment (pankajgoyal2) [inline config/tcs_config.toml]: remove the unwanted commented code. and why there is change in spid?

File: config/tcs_config.toml [hunk 2]
```diff
@@ -50,11 +51,13 @@ ias_url = "https://api.trustedservices.intel.com/sgx/dev"
 # Proxy for https. Leave commented out for direct Internet connections or
 # uncomment and change to your corporate proxy.
 #https_proxy = "http://your-proxy:your-port/"
+https_proxy = "http://proxy-chain.intel.com:912"
```
Comment (pankajgoyal2) [inline config/tcs_config.toml]: remove hard-coding intel proxies

File: config/tcs_config.toml [hunk 3]
```diff
@@ -50,11 +51,13 @@ ias_url = "https://api.trustedservices.intel.com/sgx/dev"
...
 
 # IAS API key is a 32-digit hex string subscription key used for authentication
 # of requests submitted to the IAS server. Obtain the key by subscribing in
 # the portal https://api.portal.trustedservices.intel.com/
-ias_api_key = "112233445566778899aabbccddeeff00"
+#ias_api_key = "112233445566778899aabbccddeeff00"
+ias_api_key = "7e3d31484faa4c87b5d62006f4619343"
```
Comment (pankajgoyal2) [inline config/tcs_config.toml]: remove the unwanted commented code. and why there is change in ias_api_key?

File: docker-compose-sgx.yaml [hunk 4]
```diff
@@ -16,32 +16,48 @@ version: '3.5'
 
 services:
   tcf:
```
Comment (pankajgoyal2) [inline docker-compose-sgx.yaml]: change the service name to "avalon-shell"

File: ci/tcf-tests.yaml [hunk 5]
```diff
@@ -15,23 +15,41 @@
...
-        && sleep 5 \
-        && ./tools/run_tests.sh -l avalon-listener \""
+        ./tools/run_tests.sh -l avalon-listener \""
     depends_on:
       - avalon-lmdb
+      - avalon-enclave-manager
       - avalon-listener
 
+  avalon-enclave-manager:
+    container_name: avalon-enclave-manager
```
Comment (pankajgoyal2) [inline ci/tcf-tests.yaml]: delete container_name parameter

File: docker-compose.yaml [hunk 6]
```diff
@@ -16,30 +16,43 @@ version: '3.5'
 
 services:
   tcf:
```
Comment (pankajgoyal2) [inline docker-compose.yaml]: change the service name to "avalon-shell"

File: ci/tcf-tests.yaml [hunk 7]
```diff
@@ -15,23 +15,41 @@
...
         - http_proxy
         - https_proxy
         - no_proxy
-    working_dir: "/project/TrustedComputeFramework/"
+    working_dir: "/project/avalon/"
     entrypoint: "bash -c \"\
-        source scripts/tcs_startup.sh -y -l http://avalon-lmdb:9090 \
-        && sleep 5 \
-        && ./tools/run_tests.sh -l avalon-listener \""
+        ./tools/run_tests.sh -l avalon-listener \""
```
Comment (pankajgoyal2) [inline ci/tcf-tests.yaml]: in avalon-tests service, you are running tools/run_tests.sh file. This service uses docker/Dockerfile.tcf-dev which compiles the entire Avalon code through rebuild.sh. Why do we need to compile the entire source code? Please do the clean-up of docker/Dockerfile.tcf-dev

File: enclave_manager/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+WORKDIR /project/avalon/sdk
+
+RUN python3 setup.py bdist_wheel && pip3 install dist/*.whl
+
+WORKDIR /project/avalon/enclave_manager
+
+RUN echo "Building avalon enclave manager" \
+  && make \
+  && make install
```
Comment (pankajgoyal2) [inline enclave_manager/Dockerfile]: Try to follow multistage Dockerfile in order to reduce the size of final docker image. You can follow https://github.com/hyperledger/avalon/blob/1b7422804c768eabec0635bba564bd4e6803b3a4/shared_kv_storage/Dockerfile or https://github.com/hyperledger/avalon/blob/master/listener/Dockerfile

File: config/tcs_config.toml (hunk 1 above)
Comment (danintel) [reply config/tcs_config.toml]: The `spid` is just an example that needs to be changed.  Do not put a real service provider ID here.

File: config/tcs_config.toml (hunk 2 above)
Comment (danintel) [reply config/tcs_config.toml]: Do not use an internal Intel proxy value here.  Keep the original example.

File: config/tcs_config.toml (hunk 3 above)
Comment (danintel) [reply config/tcs_config.toml]: Do not use a real IAS API key here.  Keep the original example.

File: config/tcs_config.toml (hunk 1 above)
Comment (manju956) [reply config/tcs_config.toml]: I will keep the original dummy value and delete real spid value.

File: config/tcs_config.toml (hunk 2 above)
Comment (manju956) [reply config/tcs_config.toml]: Ok

File: config/tcs_config.toml (hunk 3 above)
Comment (manju956) [reply config/tcs_config.toml]: I will keep the original dummy value and delete real IAS api key

File: docker-compose-sgx.yaml (hunk 4 above)
Comment (manju956) [reply docker-compose-sgx.yaml]: Ok

File: docker-compose.yaml (hunk 6 above)
Comment (manju956) [reply docker-compose.yaml]: Ok

File: ci/tcf-tests.yaml (hunk 5 above)
Comment (manju956) [reply ci/tcf-tests.yaml]: Ok

File: ci/tcf-tests.yaml (hunk 7 above)
Comment (manju956) [reply ci/tcf-tests.yaml]: This is done because we don't have separate dockerfile for shell/client service. I understand that we are copying and compiling whole source code which is not necessary. But i want to avoid make changes to shell service and mix it up with this PR. I could raise a separate PR to address this.

Comment (pankajgoyal2) [reply ci/tcf-tests.yaml]: I understand, dockerfile for shell/client can be done in seperate PR. But the dockerfile used should be updated (remove unwanted stuff from there). This will make it faster.

**Comment by pankajgoyal2** [comment]
Jenkins job is failing...please address this

**Comment by manju956** [comment]
> please address thi

I am looking into the issue

File: ci/tcf-tests.yaml (hunk 7 above)
Comment (manju956) [reply ci/tcf-tests.yaml]: Ok, i will optimize the dockerfile.

File: docker-compose-sgx.yaml [hunk 9]
```diff
@@ -15,33 +15,49 @@
...
-        - MAKECLEAN=${MAKECLEAN:-1}
     volumes:
       # Below volume mappings are required for DISPLAY settings by heart disease GUI client
       - /tmp/.X11-unix:/tmp/.X11-unix
       - ~/.Xauthority:/root/.Xauthority
+    working_dir: "/project/avalon"
+    entrypoint: bash -c "tail -f /dev/null"
+    stop_signal: SIGKILL
+    depends_on:
+      - avalon-lmdb
```
Comment (manojgop) [inline docker-compose-sgx.yaml]: Is avalon-shell is used to run clients ? Then it should depend on other avalon components like listener and enclave manager

File: ci/docker-compose-build.yaml [hunk 10]
```diff
@@ -15,16 +15,29 @@
...
+  avalon-shell:
+    image: avalon-shell:${ISOLATION_ID}
     build:
       context: ..
-      dockerfile: docker/Dockerfile.tcf-dev
+      dockerfile: docker/Dockerfile
       args:
         - http_proxy
         - https_proxy
         - no_proxy
```
Comment (manojgop) [inline ci/docker-compose-build.yaml]: Is avalon-shell is used to run clients ? Then shell container should be created after  starting avalon components like listener and enclave manager. Add a dependency.

File: docker/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+#
+# Build:
+#   $ docker build docker -f docker/Dockerfile.tcf-dev -t avalon-shell-dev
+#   if behind a proxy, you might want to add also below options
+#   --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy --build-arg ftp_proxy=$ftp_proxy
+#
+# Run:
+#   $ cd avalon
+#   $ docker run -v $(pwd):/project/avalon -it avalon-shell-dev /bin/bash
+#   (to run with SGX HW, add options '--device=/dev/isgx -v /var/run/aesmd:/var/run/aesmd ')
```
Comment (manojgop) [inline docker/Dockerfile]: Should SGX device be here. If not, remove SGX HW related comments

File: docker/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+
+FROM ubuntu:bionic as base_image
+
+# Ignore timezone prompt in apt
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Add necessary packages
+RUN apt-get update \
+ && apt-get install -y -q \
+    build-essential \
```
Comment (manojgop) [inline docker/Dockerfile]: Is build-essential package required in base_image? You can possibly move it to build_image

File: docker/Dockerfile [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+    ca-certificates \
+    pkg-config \
```
Comment (manojgop) [inline docker/Dockerfile]: Is pkg-config package required in base_image? You can possibly move it to build_image

File: docker/Dockerfile [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+# Install setuptools,py-solc-x and web3 packages using pip because
+# these are not available in apt repository.
+# Install nose2 for running tests.
+RUN pip3 install --upgrade setuptools json-rpc web3 py-solc-x nose2
+
+WORKDIR /tmp
+# Build ("Untrusted") OpenSSL, then build ("trusted") SGX OpenSSL
+# Note: The latter will compile in HW or SIM mode depending on the
+# availability of /dev/isgx and /var/run/aesmd/aesm.socket
+RUN OPENSSL_VER=1.1.1d \
```
Comment (manojgop) [inline docker/Dockerfile]: OpenSSL is built in many components. We should try creating a separate docker image for openSSL and reuse it in other components to reduce overall build time for Avalon. Could be done in a separate PR as well

File: docker/Dockerfile [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+  && mkdir -p build \
+  && cd build \
+  && cmake .. -DUNTRUSTED_ONLY=1 \
+  && make
+
+WORKDIR /project/avalon/common/python
+
+RUN echo "Building Avalon Common Python\n" \
+  && make \
+  && make install
```
Comment (manojgop) [inline docker/Dockerfile]: installing wheel file (via make install) should be done in final image. Refer shared_kv_storage Dockerfile. https://github.com/hyperledger/avalon/blob/master/shared_kv_storage/Dockerfile

File: docker/Dockerfile [hunk 16]
```diff
@@ -0,0 +1,182 @@
...
+
+WORKDIR /project/avalon/sdk
+
+RUN echo "Building Avalon SDK\n" \
+  && python3 setup.py bdist_wheel \
+  && pip3 install dist/*.whl
```
Comment (manojgop) [inline docker/Dockerfile]: https://github.com/hyperledger/avalon/blob/master/shared_kv_storage/Dockerfile

File: enclave_manager/Dockerfile [hunk 17]
```diff
@@ -14,80 +14,65 @@
...
-    python3-dev \
-    virtualenv \
-    python3-pip \
     python3-toml \
     python3-requests \
     python3-colorlog \
     python3-twisted \
+ && apt-get -y -q upgrade \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
```
Comment (manojgop) [inline enclave_manager/Dockerfile]: build-essential and  pkg-config could be moved to build_image

File: docker-compose-sgx.yaml (hunk 9 above)
Comment (manju956) [reply docker-compose-sgx.yaml]: Yes, its used to run clients and tests. Client depends on the db to be up and running along with enclave manager and listener. But adding enclave manager and listener as dependencies make sense because they are dependent on lmdb service to be up. I will make the change.

File: ci/docker-compose-build.yaml (hunk 10 above)
Comment (manju956) [reply ci/docker-compose-build.yaml]: Yes, its used to run clients and tests. Client depends on the db to be up and running along with enclave manager and listener. But adding enclave manager and listener as dependencies make sense because they are dependent on lmdb service to be up. I will make the change.

File: docker/Dockerfile (hunk 11 above)
Comment (manju956) [reply docker/Dockerfile]: Now client don't depend on SGX anymore. I will remove the comments.

File: docker/Dockerfile (hunk 14 above)
Comment (manju956) [reply docker/Dockerfile]: This change need to be made in separate PR.

File: docker/Dockerfile (hunk 15 above)
Comment (manju956) [reply docker/Dockerfile]: I referred this from listener dockerfile. I will install the wheel package in the final image.

File: docker/Dockerfile (hunk 16 above)
Comment (manju956) [reply docker/Dockerfile]: Ok

File: docker/Dockerfile (hunk 13 above)
Comment (manju956) [reply docker/Dockerfile]: Ok

File: docker/Dockerfile (hunk 12 above)
Comment (manju956) [reply docker/Dockerfile]: Ok

File: enclave_manager/Dockerfile [hunk 18]
```diff
@@ -137,61 +121,136 @@ RUN OPENSSL_VER=1.1.1d \
...
+
+ENV http_proxy=$http_proxy
+ENV https_proxy=$https_proxy
+ENV TCF_HOME=/project/avalon
+
+WORKDIR /project/avalon/
+
+COPY --from=build_image /usr/local/ /usr/local
+COPY --from=build_image /project/avalon/config/tcs_config.toml /project/avalon/config/
+COPY --from=build_image /project/avalon/tc/sgx/trusted_worker_manager/enclave/build/ /project/avalon/tc/sgx/trusted_worker_manager/enclave/build
```
Comment (manojgop) [inline enclave_manager/Dockerfile]: We don;t need to copy entire build artifacts. We need to copy only required files like shared library and executable ?

File: enclave_manager/Dockerfile [hunk 19]
```diff
@@ -137,61 +121,114 @@ RUN OPENSSL_VER=1.1.1d \
...
+  && make
 
-RUN if [ "$MAKECLEAN" != 0 ]; then echo "Clean build" && make clean; fi \
- && make \
- && echo "BUILD SUCCESS"
+WORKDIR /project/avalon/common/python
+
+RUN make
+
+#ENV PATH=$PATH:/project/avalon/examples/enclave_manager
```
Comment (manojgop) [inline enclave_manager/Dockerfile]: Please remove commented line

File: enclave_manager/Dockerfile [hunk 20]
```diff
@@ -137,61 +121,114 @@ RUN OPENSSL_VER=1.1.1d \
...
+
+WORKDIR /project/avalon/sdk
+
+RUN python3 setup.py bdist_wheel
```
Comment (manojgop) [inline enclave_manager/Dockerfile]: The Makefile in avalon SDK should do above step

File: enclave_manager/Dockerfile [hunk 21]
```diff
@@ -137,61 +121,114 @@ RUN OPENSSL_VER=1.1.1d \
...
+# Build Final image and install dependent modules
+FROM base_image as final_image
+
+ENV http_proxy=$http_proxy
+ENV https_proxy=$https_proxy
+ENV TCF_HOME=/project/avalon
+
+WORKDIR /project/avalon/
+
+COPY --from=build_image /usr/local/ /usr/local
```
Comment (manojgop) [inline enclave_manager/Dockerfile]: We can remove above line since wheel package is installed in final image

File: enclave_manager/Dockerfile [hunk 22]
```diff
@@ -137,61 +121,114 @@ RUN OPENSSL_VER=1.1.1d \
...
+COPY --from=build_image /project/avalon/config/tcs_config.toml /project/avalon/config/
+COPY --from=build_image /project/avalon/tc/sgx/trusted_worker_manager/enclave/build/lib/*.so /project/avalon/tc/sgx/trusted_worker_manager/enclave/build/lib/
+COPY --from=build_image /project/avalon/tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/build/lib/*.so /project/avalon/tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/build/lib/
+COPY --from=build_image /opt/intel/ /opt/intel
```
Comment (manojgop) [inline enclave_manager/Dockerfile]: What is there in /opt/intel ? Please add a comment

File: enclave_manager/Dockerfile [hunk 23]
```diff
@@ -137,61 +121,114 @@ RUN OPENSSL_VER=1.1.1d \
...
+RUN openssl genrsa -3 -out $TCF_HOME/enclave.pem 3072
+
+WORKDIR /project/avalon/common/sgx_workload
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/examples/apps
```
[the remark this answers is not in the document]
Comment (manju956) [reply enclave_manager/Dockerfile]: Because apps need to be linked to enclave shared library.

File: enclave_manager/Dockerfile (hunk 19 above)
Comment (manju956) [reply enclave_manager/Dockerfile]: Ok

File: enclave_manager/Dockerfile (hunk 20 above)
Comment (manju956) [reply enclave_manager/Dockerfile]: Ok, will switch to make command

File: enclave_manager/Dockerfile (hunk 22 above)
Comment (manju956) [reply enclave_manager/Dockerfile]: /opt/intel is the place holder to store sgxssl, sgxsdk. I shall add the comment.

File: enclave_manager/Dockerfile (hunk 21 above)
Comment (manju956) [reply enclave_manager/Dockerfile]: This command is to copy the python modules that are downloaded during building stage. these modules are the run-time dependency for Avalon python modules. Hence copying them to final image.

File: docker/Dockerfile [hunk 24]
```diff
@@ -0,0 +1,160 @@
...
+#Environment setup
+ENV http_proxy=$http_proxy
+ENV https_proxy=$https_proxy
+ENV TCF_HOME=/project/avalon
+# Set display, replace value with [IP address of host]:0
+ENV DISPLAY=:0
+
+WORKDIR /project/avalon/
+
+COPY --from=build_image /usr/local/ /usr/local
```
[the remark this answers is not in the document]
Comment (manju956) [reply docker/Dockerfile]: This command is to copy the python modules that are downloaded during building stage. these modules are the run-time dependency for Avalon python modules. Hence copying them to final image.

File: enclave_manager/Dockerfile (hunk 17 above)
Comment (manju956) [reply enclave_manager/Dockerfile]: Done

File: docker/Dockerfile [hunk 25]
```diff
@@ -0,0 +1,165 @@
...
+
+# Installing solidity compiler
+RUN add-apt-repository ppa:ethereum/ethereum \
+   && apt-get update \
+   && apt-get install solc
+
+# Install py-solc-x and web3 packages using pip because
+# these are not available in apt repository.
+# Install nose2 for running tests.
+RUN pip3 install --upgrade setuptools json-rpc web3 py-solc-x nose2
```
Comment (manojgop) [inline docker/Dockerfile]: Do you need setuptools here. setuptools is installed in build image

File: docker/Dockerfile [hunk 26]
```diff
@@ -0,0 +1,165 @@
...
+#Environment setup
+ENV http_proxy=$http_proxy
+ENV https_proxy=$https_proxy
+ENV TCF_HOME=/project/avalon
+# Set display, replace value with [IP address of host]:0
+ENV DISPLAY=:0
+
+WORKDIR /project/avalon/
+
+#COPY --from=build_image /usr/local/ /usr/local
```
Comment (manojgop) [inline docker/Dockerfile]: remove commented line

File: docker/Dockerfile [hunk 27]
```diff
@@ -0,0 +1,165 @@
...
+# ------------------------------------------------------------------------------
+
+# Description:
+#   Builds the environment needed to build Avalon shell.
+#
+#  Configuration (build) parameters
+#  - proxy configuration: https_proxy http_proxy ftp_proxy
+#
+# Build:
+#   $ docker build docker -f docker/Dockerfile.tcf-dev -t avalon-shell-dev
```
Comment (manojgop) [inline docker/Dockerfile]: docker/Dockerfile.tcf-dev -> docker/Dockerfile

File: docker/Dockerfile (hunk 25 above)
Comment (manju956) [reply docker/Dockerfile]: Not needed, its part of build image. Removed it.

File: docker/Dockerfile (hunk 26 above)
Comment (manju956) [reply docker/Dockerfile]: Done

File: docker/Dockerfile (hunk 27 above)
Comment (manju956) [reply docker/Dockerfile]: Done

File: enclave_manager/Dockerfile (hunk 23 above)
[the remark this answers is not in the document]
Comment (manojgop) [reply enclave_manager/Dockerfile]: Please add a comment that apps directory consists of SGX workloads which needs to be build and linked to enclave library

File: bin/run_lint [hunk 28]
```diff
@@ -267,7 +267,7 @@ rv=0
 #                                           pylint |    cppcheck
 export PYTHONPATH=$PYTHONPATH:$top_dir/examples/common/python
 lint_module tests                    "$SINCE" 1  0   0  0 || rv=1
-lint_module examples/enclave_manager "$SINCE" 1  0   0  0 || rv=1
+lint_module enclave_manager          "$SINCE" 1  0   0  0 || rv=1
 lint_module examples/common          "$SINCE" 1  0   0  0 || rv=1
```
Comment (manojgop) [inline bin/run_lint]: Shouldn't we change example/common -> common/python. May be in a separate PR

File: docker/Dockerfile [hunk 29]
```diff
@@ -0,0 +1,164 @@
...
+ENV DISPLAY=:0
+
+WORKDIR /project/avalon/
+
+COPY --from=build_image /project/avalon/config/tcs_config.toml /project/avalon/config/
+COPY --from=build_image /project/avalon/examples/apps/ /project/avalon/examples/apps
+COPY --from=build_image /project/avalon/tests /project/avalon/tests
+COPY --from=build_image /project/avalon/sdk/avalon_sdk/tcf_connector.toml /project/avalon/sdk/avalon_sdk/
+COPY --from=build_image /project/avalon/sdk/avalon_sdk/ethereum/contracts/WorkerRegistryList.sol /project/avalon/sdk/avalon_sdk/ethereum/contracts/
+COPY --from=build_image /project/avalon/sdk/avalon_sdk/ethereum/contracts/WorkerRegistry.sol /project/avalon/sdk/avalon_sdk/ethereum/contracts/
```
Comment (manojgop) [inline docker/Dockerfile]: Above two lines of copying contracts can be removed . Were you facing any issue if you don;t copy these files.

Comment (manju956) [reply docker/Dockerfile]: Its needed because these two contract files are used while initilaizing ethereum_wrapper as part of sdk.

File: bin/run_lint (hunk 28 above)
Comment (manju956) [reply bin/run_lint]: Yes, will do it in separate PR

File: enclave_manager/Dockerfile (hunk 23 above)
[the remark this answers is not in the document]
Comment (manju956) [reply enclave_manager/Dockerfile]: Done

File: docker/Dockerfile (hunk 29 above)
Comment (manojgop) [reply docker/Dockerfile]: @rranjan3 Please remove this dependency in a separate PR

File: docker/Dockerfile [hunk 30]
```diff
@@ -0,0 +1,170 @@
+# Copyright 2019 Intel Corporation
```
Comment (pankajgoyal2) [inline docker/Dockerfile]: 2019 -> 2020

Comment (manju956) [reply docker/Dockerfile]: This is a old file renamed to Dockerfile in this PR. Hence update is not needed.

Comment (manju956) [reply docker/Dockerfile]: Ok

File: docker/Dockerfile [hunk 31]
```diff
@@ -0,0 +1,188 @@
...
+    libprotobuf-dev \
+    libssl-dev \
+    python3-dev \
+    swig \
+    tar
+
+RUN pip3 install --upgrade setuptools
+
+# Copy openssl build artifacts from openssl_image
+ENV OPENSSL_VER=1.1.1d
```
Comment (pankajgoyal2) [inline docker/Dockerfile]: Remove the above command. Its not needed.

Comment (manju956) [reply docker/Dockerfile]: Done

Comment (manju956) [reply docker/Dockerfile]: Done

**Comment by pankajgoyal2** [comment]
@manju956 I observed that in the latest commit, you have introduced a new change (renaming common/python to common/crypto) which seem not to be related to this PR. Please revert this change and raise a seperate PR if at all that is required.

**Comment by manju956** [comment]
> Looks like some new changes related to crypto was introduced accidentally. Please revert it and resubmit new commit

Accidentally, few of the file changes to crypto pushed to this PR. I am reverting it.

**Comment by manju956** [comment]
> @manju956 I observed that in latest commit, you have introduced a new change (renaming common/python to common/crypto) which seem not be related to this PR. Please revert this change and raise a seperate PR if at all that is required.

I am reverting the accidental file commit.

File: enclave_manager/setup.py [hunk 32]
```diff
@@ -106,12 +106,14 @@
       url = 'https://github.com/hyperledger/avalon',
       packages = find_packages(),
       install_requires = [
-          'requests',
-          'toml',
+          'requests'
```
Comment (pankajgoyal2) [inline enclave_manager/setup.py]: line no 102: change name to avalon_enclave_manager

File: enclave_manager/tcf_enclave_manager/enclave_manager.py [hunk 33]
```diff
@@ -22,8 +22,8 @@
 import time
 import random
 
-import sgx_work_order_request as work_order_request
-import tcf_enclave_helper as enclave_helper
+import tcf_enclave_manager.sgx_work_order_request as work_order_request
```
Comment (pankajgoyal2) [inline enclave_manager/tcf_enclave_manager/enclave_manager.py]: change directory/file names to avalon_* from tcf_*

File: enclave_manager/Dockerfile [hunk 34]
```diff
@@ -165,65 +157,116 @@ RUN ldconfig \
...
-# Set display, replace value with [IP address of host]:0
-ENV DISPLAY=:0
+ENV TCF_ENCLAVE_CODE_SIGN_PEM="$TCF_HOME/enclave.pem"
+
+RUN openssl genrsa -3 -out $TCF_HOME/enclave.pem 3072
 
-WORKDIR /project/TrustedComputeFramework/tools/build
+WORKDIR /project/avalon/tools/build
 
 RUN make lint && echo "LINT SUCCESS"
```
Comment (manojgop) [inline enclave_manager/Dockerfile]: Remove run lint from enclave manager Dockerfile as currently lint checks all directories in Avalon

File: enclave_manager/Dockerfile [hunk 35]
```diff
@@ -165,65 +157,116 @@ RUN ldconfig \
  && rm -rf /tmp/intel-sgx-ssl \
  && echo "SGX_SSL=/opt/intel/sgxssl" >> /etc/environment
 
-COPY . /project/TrustedComputeFramework
+COPY . /project/avalon
```
Comment (manojgop) [inline enclave_manager/Dockerfile]: We have to copy only required directories for building enclave manager

File: enclave_manager/tcf_enclave_manager/enclave_manager.py (hunk 33 above)
Comment (manju956) [reply enclave_manager/tcf_enclave_manager/enclave_manager.py]: Will address renaming filename from tcf to avalon in another PR.

File: enclave_manager/Dockerfile (hunk 34 above)
Comment (manju956) [reply enclave_manager/Dockerfile]: Ok

File: enclave_manager/tcf_enclave_manager/enclave_manager.py (hunk 33 above)
Comment (pankajgoyal2) [reply enclave_manager/tcf_enclave_manager/enclave_manager.py]: Renaming "tcf" to "avalon" in enclave-manager context makes sense to handle that in this PR itself. There might be other such instances in other parts of the code which can be picked up later. Same way has been done in other modules of avalon also.

File: enclave_manager/Dockerfile [hunk 36]
```diff
@@ -165,65 +153,112 @@ RUN ldconfig \
...
-RUN cd /project/TrustedComputeFramework/sdk \
-   && python setup.py bdist_wheel \
-   && pip3 install dist/*.whl
-
-# Install fabric python sdk
-RUN pip3 install virtualenv \
-   && pip3 install git+https://github.com/hyperledger/fabric-sdk-py.git
 
 # Environment setup
-
```
Comment (manojgop) [inline enclave_manager/Dockerfile]: Are these proxy settings no more needed for enclave manager. Will it work in case of h/w mode if enclave manger needs to contact IAS server ?

File: BUILD.md [hunk 37]
```diff
@@ -52,7 +52,7 @@ Follow the instructions below to execute a Docker-based build and execution.
 3. On a successful run, you should see the message `BUILD SUCCESS`
    followed by a repetitive message `Enclave manager sleeping for 10 secs`
 4. Open a Docker container shell using following command
-   `sudo docker exec -it tcf bash`
+   `sudo docker exec -it avalon-shell bash`
```
Comment (manojgop) [inline BUILD.md]: Please make the documenttaion changes in below files as well.
docs/TestingFabricProxyModel.md
docs/BesuProxyModel.md
docs/workload-tutorial/README.md

**Comment by manju956** [comment]
> Please make the relevant changes in tools/rebuild.sh as well.

Done

File: BUILD.md (hunk 37 above)
Comment (manju956) [reply BUILD.md]: Done

File: enclave_manager/Dockerfile (hunk 36 above)
Comment (manju956) [reply enclave_manager/Dockerfile]: Yes, its not needed, because IAS server proxy will be picked from configuration file.
